### PR TITLE
added batch deleter unit test

### PR DESCRIPTION
**Pull request by saurav-malani**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: regulation-worker/internal/delete/batch/batch_test.go [hunk 1]
```diff
@@ -57,12 +71,67 @@ func TestBatchDelete(t *testing.T) {
...
+				}
+				return nil
+			})
+			if err != nil {
+				panic(err)
+			}
+			sort.Strings(goldenFilesList)
+			require.Equal(t, len(goldenFilesList), len(cleanedFilesList), "actual number of files in destination bucket different than expected")
+			for i := 0; i < len(goldenFilesList); i++ {
+				goldenFilePtr, err := os.OpenFile(goldenFilesList[i], os.O_RDWR, 0644)
```
Comment (lvrach) [inline regulation-worker/internal/delete/batch/batch_test.go:109]: Any reason for using `OpenFile` ? We can simplify things by:
```suggestion
				goldenFilePtr, err := os.Open(goldenFilesList[i])
```

File: regulation-worker/internal/delete/batch/batch_test.go [hunk 2]
```diff
@@ -57,12 +71,67 @@ func TestBatchDelete(t *testing.T) {
...
+				if err != nil {
+					panic(err)
```
Comment (lvrach) [inline regulation-worker/internal/delete/batch/batch_test.go:111]: Prefer:
```suggestion
					t.Fatal(err)
```

or

```suggestion
     require.NoError(err)
```

File: regulation-worker/internal/delete/batch/batch_test.go [hunk 3]
```diff
@@ -71,3 +140,120 @@ func TestBatchDelete(t *testing.T) {
...
+//creates a tmp directory & copy all the content of testData in it, to use it as mockBucket & store it in mockFileManager struct.
+func (ff mockFileManagerFactory) New(settings *filemanager.SettingsT) (filemanager.FileManager, error) {
+	//create tmp directory
+	//parent directory of all the temporary files created/downloaded in the process of deletion.
+	tmpDirPath, err := os.MkdirTemp("", "")
+	if err != nil {
+		panic(err)
+	}
+	//copy all content of testData in the tmp directory
+	_, err = exec.Command("cp", "-r", "./mockBucket", tmpDirPath).Output()
```
Comment (lvrach) [inline regulation-worker/internal/delete/batch/batch_test.go:156]: A good idea would be to move this to a constant ?

File: regulation-worker/internal/delete/batch/batch_test.go [hunk 4]
```diff
@@ -71,3 +140,120 @@ func TestBatchDelete(t *testing.T) {
...
+	fileName := ""
+	if len(prefixes) > 0 {
+		fileName = strings.Join(prefixes[:], "/") + "/"
+	}
+	fileName += splitFileName[len(splitFileName)-1]
+	//copy the content of file to mockBucektLocation+fileName
+	finalFileName := fmt.Sprintf("%s%s%s", fm.mockBucketLocation, "/", fileName)
+	uploadFilePtr, err := os.OpenFile(finalFileName, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		fmt.Println("error: ", err, " while opening file: ", fileName)
```
Comment (lvrach) [inline regulation-worker/internal/delete/batch/batch_test.go:184]: Why not returning an error ? The following command will fail for sure

File: regulation-worker/internal/delete/batch/batch_test.go [hunk 5]
```diff
@@ -71,3 +140,120 @@ func TestBatchDelete(t *testing.T) {
...
+}
+
+//Given a file name download & simply save it in the given file pointer.
+func (fm *mockFileManager) Download(outputFilePtr *os.File, location string) error {
+	finalFileName := fmt.Sprintf("%s%s%s", fm.mockBucketLocation, "/", location)
+
+	uploadFilePtr, err := os.OpenFile(finalFileName, os.O_RDWR, 0644)
+	if err != nil {
+		fmt.Println("error: ", err, " while opening file: ", location)
+		return nil
```
Comment (lvrach) [inline regulation-worker/internal/delete/batch/batch_test.go:205]: Again error could be propagated.

**Review by lvrach — CHANGES_REQUESTED** [review]
Overall implementation is good, requesting changes for error handling

**Review by chandumlg — APPROVED** [review]

**Review by lvrach — APPROVED** [review]

**Pull request merged**